utils.go

```diff
@@ -212,3 +213,19 @@ func SplitCells(cells []Cell, r rune) [][]Cell {
 	}
 	return splitCells
 }
+
+type CellWithX struct {
+	X    int
+	Cell Cell
+}
+
+func BuildCellChannel(cells []Cell) chan CellWithX {
+	c := make(chan CellWithX, len(cells))
+	index := 0
+	for _, cell := range cells {
+		c <- CellWithX{X: index, Cell: cell}
+		index += rw.RuneWidth(cell.Rune)
+	}
+	close(c)
+	return c
+}
```

Comment:
This is an interesting usage of channels but would it make more sense to just return a `[]CellWithX`?

Reply:
Yes! U r right!
I mean I'm a newbie of golang……I'll change this to return just a `[]CellWithX`